cat-home/src/test/java/com/dianping/cat/report/task/TaskHelperTest.java

```diff
@@ -65,4 +76,84 @@ public void testTodayZero() {
 		Assert.assertEquals(cal.getTimeInMillis(), todayZero.getTime() + 1L);
 	}
 
+    @Test
+    public void testTodayZero2() {
+        Assert.assertEquals(new Date(1562713200000L),
+                TaskHelper.todayZero(null));
+        Assert.assertEquals(new Date(1562713200000L),
+                TaskHelper.todayZero(new Date()));
+    }
+
+    @Test
+    public void testJoin1() {
+
+        final int[] array = {1, -8, -7, -7, -7, -7, -7, -253_968, -7, -24, -24,
+                -2147475464, 4088, -2147483647, 9, 33505273, 1, -8};
+
+
+        Assert.assertNull(TaskHelper.join((int[]) null, '\u0000',
+                -160729767, -159952269));
+        Assert.assertNull(TaskHelper.join((Number[]) null, '\u0000',
+                -1, 2147483647));
+        Assert.assertNull(TaskHelper.join((double[]) null, '\u0000',
+                -402653175, 1669333001));
+
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000'));
+        Assert.assertEquals("", TaskHelper.join(new double[]{5.1}, '\u0000'));
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000',
+                -2147483648, 0));
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000',
+                1816002434, -1402208264));
+        Assert.assertEquals("", TaskHelper.join(new Number[]{null}, '\u0000',
+                2147483647, 1073742086));
+        Assert.assertEquals("", TaskHelper.join(new double[]{
+                        5.1, 3.5}, '\u0000',
+                -450887681, -1299185664));
+        Assert.assertEquals("2.1\u00004.5",
+                TaskHelper.join(new double[]{2.1, 4.5}, '\u0000'));
+        Assert.assertEquals("", TaskHelper.join(new double[]
+                {5.1, 9.2}, '\u0000', 1197600769, 964623358));
```

Comment:
Same as previous (Or at least make it consistent with example above) as one was before `{` and one was after

Reply:
done